packages/rlp/CHANGELOG.md

```diff
@@ -27,7 +27,26 @@ RLP.encode(1)
 
 ### Uint8Array
 
-Buffers were replaced in favor of using Uint8Arrays for greater compatibility with browsers.
+Buffers were replaced in favor of using Uint8Arrays for improved performance and greater compatibility with browsers.
+
+When upgrading from rlp v2 to v3, you must convert your Buffers to Uint8Arrays before passing in. To help, two new utility methods were added to `ethereumjs-util v7.1.4`: `arrToBufArr` and `bufArrToArr`. These will recursively step through your arrays to replace Buffers with Uint8Arrays, or vise versa.
+
+Example:
+
+```typescript
+// Old, rlp v2
+import * as rlp from 'rlp'
+const bufArr = [Buffer.from('123', 'hex'), Buffer.from('456', 'hex')]
+const encoded = rlp.encode(bufArr)
+const decoded = rlp.decode(encoded)
+
+// New, rlp v3
+import RLP from 'rlp'
+const encoded: Uint8Array = RLP.encode(bufArrToArr(bufArr))
+const encodedAsBuffer = Buffer.from(encoded)
+const decoded: Uint8Array[] = RLP.decode(encoded)
+const decodedAsBuffers = arrToBufArr(decoded)
+```
```

Comment:
Just a note: when we release v3 we should add this to the README to for greater visibility (alternative we can also already add now with a "(not yet released)" addition after "v3" and then remove on v3 release.

Reply:
sounds good, I already updated the README but can add a section for Buffer compatibility since the topic isn't directly discussed there.

Reply:
> Hmm, can you please test this new typing of the conversion methods on the v3 PR on #1630? I just injected the typings in there and get a lot of "No overload matches this call." errors already in the Util library when trying to compile.

Just tried it out, this was the one place that error'd out: (this is because the version on that branch uses `any` so it basically skips type checks)

> Type 'Decoded' is missing the following properties from type '(Uint8Array | NestedUint8Array)[]': length, pop, push, concat, and 32 more.
>  `const values = arrToBufArr(RLP.decode(Uint8Array.from(serialized)))`

This is because we don't know what type `RLP.decode` outputs, so this can be fixed with:

`const values = arrToBufArr(RLP.decode(Uint8Array.from(serialized)) as Uint8Array[])`

then we have to cast `as Buffer[]` for `fromValuesArray(values: Buffer[])` because `NestedBufferArray` !== `Buffer[]`

which I think is okay. So:

`const values = arrToBufArr(RLP.decode(Uint8Array.from(serialized)) as Uint8Array[]) as Buffer[]`

a little ugly but better than using `any`?

My overload typings above just help if you pass in a plain non-array Uint8Array or Buffer you get the non-array version back, so there should be less need for manual overrides in those cases to cast.

Reply:
have added a Buffer compatibility section to the readme here: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1648/files#diff-68eff8537817899e4ff13ddb30d65da860ae9de825c4426c5551b3bf5ede53a6R35 let me know if it looks good